enumfields/enums.py

```diff
@@ -39,6 +41,33 @@ def __new__(mcs, name, bases, attrs):
 
 class ChoiceEnum(ChoiceEnumMeta('ChoiceEnum', (BaseEnum,), _EnumDict())):
 
+    @classmethod
+    def get_allowed_previous_choices(cls, current_choice):
+        return cls._get_previous_map().get(current_choice) or []
+
+    @classmethod
+    @lru_cache(maxsize=1)
+    def get_initial_choices(cls):
+        return [
+            choice for choice in cls
+            if not hasattr(choice, 'initial') or choice.initial
+        ]
+
+    @classmethod
+    @lru_cache(maxsize=1)
+    def _get_previous_map(cls):
+        previous_map = {choice.name: [] for choice in cls}
```

Comment:
better solution should be if next property of choice will transform to the real enum choices during enum inicialization

Reply:
I agree, it will be also less memory demanding, and you dont need any of the classmethods above.

Reply:
yes, but I deleted it because now I know the correct previous value so I can compute the transition easier way.